addons/website/static/src/client_actions/website_editor/website_editor.js

```diff
@@ -44,11 +45,29 @@ export class WebsiteEditorClientAction extends Component {
                     this.iframefallback.el.contentDocument.body.replaceWith(this.iframe.el.contentDocument.body.cloneNode(true));
                     $().getScrollingElement(this.iframefallback.el.contentDocument)[0].scrollTop = $().getScrollingElement(this.iframe.el.contentDocument)[0].scrollTop;
                 });
+
+                this.iframe.el.contentWindow.addEventListener('PUBLIC-ROOT-READY', (event) => {
+                    if (!this.websiteContext.edition) {
+                        const $wrap = $(this.iframe.el.contentDocument.getElementById('wrap'));
+                        if ($wrap.length && $wrap.html().trim() === '') {
+                            this.$welcomeMessage = $(core.qweb.render('website.homepage_editor_welcome_message'));
+                            this.$welcomeMessage.addClass('o_homepage_editor_welcome_message');
+                            this.$welcomeMessage.css('min-height', $wrap.parent('main').height() - ($wrap.outerHeight(true) - $wrap.height()));
+                            $wrap.empty().append(this.$welcomeMessage);
```

Comment:
try to avoid to use jQuery if possible

Reply:
I try to convert as much as possible, but this code was ported over from the old Edit.js widget.
I will try to convert it into plain JS sometimes in the future.

Another thing that we will do is a commit that purely marks everything that was simply moved from one file to another. cc @younn-o